### ISSUE TITLE
HTTPS support for CartoDB/MapQuest?

### ISSUE THREAD

**Issue by stuporglue** [leaflet-extras/leaflet-providers#189]
It'd be great if CartoDB and MapQuest https support could be added. I know that their HTTPS tile schemas are different, which makes it non-trivial. 

In some code where I'm currently using mapquest tiles on http and https sites I am initializing the tile layer this way: 

```
    if(window.location.protocol === 'https:'){
        subDomains = ['otile1-s','otile2-s','otile3-s','otile4-s'];
    } else{
        subDomains = ['otile1','otile2','otile3','otile4'];
    }
    return new L.TileLayer('//{s}.mqcdn.com/tiles/1.0.0/map/{z}/{x}/{y}.png',{...options here...})
```

If this could be considered as a future feature request, I'd appreciate it.

**Comment by stuporglue** [comment]
Maybe something like this would be OK? 

```
    url: '//otile{s}' + (window.location.protocol == 'https:' ? '-s' : '') + '.mqcdn.com/tiles/1.0.0/map/{z}/{x}/{y}.{ext}'
```

**Comment by jieter** [comment]
@stuporglue thanks for thinking about this issue. I think I like your proposal if we move the check to variable so it might be reused for other layers having different url schemes for http and https. Do you want to make a pull request?

thoughs @brunob?

**Comment by brunob** [comment]
Think it's nice to support more providers with https since letsencrypt.org is coming we will meet more and more https websites.

**Comment by jieter** [comment]
#195 had this implemented, but was doing to much at once. Need to revisit this.

**Comment by brunob** [comment]
Fixed by #258
